### PR TITLE
[risk=no] FakeClockConfiguration rework

### PR DESCRIPTION
**Pull request by calbach**
- rm `SpringTest`
- If a test needs a Clock, `@Import(FakeClockConfiguration.class)` on the TestConfiguration if any, else on the Test itself
- If a test needs DAOs, `@DataJpaTest` on the test as usual
- if a test needs Spring, without DAOs: `@SpringJunitConfig` on the test

### CODE REVIEW

**Comment by calbach** [comment]
If this is approved, will add a section to the Wiki describing these different scenarios

**Review by jmthibault79 — APPROVED** [review]

File: api/src/test/java/org/pmiops/workbench/actionaudit/ActionAuditEventTest.java [hunk 1]
```diff
@@ -5,9 +5,11 @@
 
 import java.time.Instant;
 import org.junit.jupiter.api.Test;
-import org.pmiops.workbench.SpringTest;
+import org.pmiops.workbench.FakeClockConfiguration;
+import org.springframework.context.annotation.Import;
 
-public class ActionAuditEventTest extends SpringTest {
+@Import(FakeClockConfiguration.class)
+public class ActionAuditEventTest {
```
[the remark this answers is not in the document]
Comment (calbach) [reply api/src/test/java/org/pmiops/workbench/actionaudit/ActionAuditEventTest.java:9]: As long as it's not autowiring anything this is fine. However, my `@Import` was a no-op in such a configuration - removed.

**Review by calbach — COMMENTED** [review]
thanks

File: api/src/test/java/org/pmiops/workbench/actionaudit/ActionAuditEventTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (calbach) [reply api/src/test/java/org/pmiops/workbench/actionaudit/ActionAuditEventTest.java:9]: found and removed 4 more such cases

**Pull request merged**